department-of-veterans-affairs/abd-vro
# EPMerge: Fix serialization issue resulting in bad request response from BIP API

## What was the problem?

`svc-bip-api` was not correctly serializing bodies containing date/timestamp fields for requests made to BIP API. Turns out that the message converter for BIP api was not using the JavaTimeModule.

Associated tickets or Slack threads:
- #2714
- Thought it was fixed in #2715 but issue still persisted (see slack [thread](https://dsva.slack.com/archives/C01BVF2A3V0/p1709842883373739))

## How does this fix it?[^1]

Uses the configured object mapper to serialize the requests with bodies for RestTemplate
 
## How to test this PR
- run unit and integration tests for `svc-bip-api` 

Run EP Merge tests as well: 
- End to End tests:
  - run this [CI](https://github.com/department-of-veterans-affairs/abd-vro/actions/workflows/ee-ep-merge-end-to-end.yml) action to verify successful integration test
  - run end to end tests locally:
    - `export BIP_CLAIM_URL=mock-bip-claims-api:20300`
    - run bgs and bip services and mocks:
      - `COMPOSE_PROFILES="bip,bgs" ./gradlew :app:dockerComposeUp`
      - `COMPOSE_PROFILES="bip,bgs" ./gradlew -p mocks :dockerComposeUp`
    - `./gradlew :domain-ee:ee-ep-merge-app:endToEndTest`

Note that the tests above use the `mock-bip-claims-api` which is VRO's best-guess based on swagger documentation of how actual BIP API works. So I also tested by deploying this commit to dev and finding a claim (id: 600145754) and making a manual request to `svc-bip-api` to create claim contentions endpoint using curl commands and putting a request into rabbitmq `bipApiExchange`. The steps outlined in #2715 can be followed to repeat the process assuming that the deployment has not been overwritten. 

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
